dlt/sources/rest_api/config_setup.py

```diff
@@ -577,16 +624,101 @@ def remove_field(response: Response, *args, **kwargs) -> Response:
     return None
 
 
+def _extract_expressions(
+    template: Union[str, Dict],
+    prefix: str = "",
+) -> List[str]:
+    """Takes a template string and extracts expressions that start with a prefix.
+    Args:
+        template (str): A string with expressions to extract
+        prefix (str): A string that marks the beginning of an expression
+    Example:
+        >>> _extract_expressions("blog/{resources.blog.id}/comments", "resources.")
+        ["resources.blog.id"]
+    """
+
+    expressions = set()
+
+    def recursive_search(value):
+        if isinstance(value, dict):
+            for key, val in value.items():
+                recursive_search(key)
```

Comment:
Could you elaborate on the use case for this? Is this the case for JSON with parametrized keys?